app/src/pages/inside/logsPage/historyLine/historyLine.jsx

```diff
@@ -94,25 +94,35 @@ export class HistoryLine extends Component {
   };
 
   loadMoreItems = () => {
+    const { historyItems } = this.props;
     if (this.state.isLoading) {
       return;
     }
     this.setState({ isLoading: true });
+    this.setState({ prevItems: historyItems.length });
     this.props.fetchHistoryItemsAction(true, this.finishLoading);
-  };
-
-  render() {
-    const { historyItems, activeItemId, changeActiveItem, intl } = this.props;
     const loadedItems = historyItems.length - DEFAULT_HISTORY_DEPTH;
     const shouldShowLoadMore =
+      this.state.prevItems !== historyItems.length &&
       historyItems.length < HISTORY_DEPTH_LIMIT &&
       loadedItems >= 0 &&
       loadedItems % NUMBER_OF_ITEMS_TO_LOAD === 0;
+    this.setState({ shouldShowLoadMore });
```

Comment:
You use `setState` triple in one place, it may cause two redundant rerenders. Please use just one to update all necessary state properties.
I also suggest to move `shouldShowLoadMore` and `prevItemsCount` to the `finishLoading` to have actual data for these properties calculation.